src/Persistance/Json/JsonFileHandler.php

```diff
@@ -109,6 +109,25 @@ public function write(string $filename, mixed $jsonable): void
         }
     }
 
+    /**
+     * Remove the file from storage.
+     *
+     * @param string $filename
+     * @return void
+     */
+    public function unlink(string $filename): void
+    {
+        // Given the path to the file using the slug
+        $slug = $this->pathTo($filename);
+        // If the list of that filename exists then
+        if ($this->exists($filename) && $filename === $slug) {
```

Comment:
Required: this won't work, but is extremely close. The comment you've put above the `if` statement does accurately describe what the if should check, but what you've written in code doesn't match it.

Reply:
I updated the code!